language/bytecode_verifier/src/verifier.rs

```diff
@@ -163,6 +163,9 @@ impl VerifiedModule {
         if errors.is_empty() {
             errors.append(&mut RecursiveStructDefChecker::new(&module).verify());
         }
+        if errors.is_empty() {
```

Comment:
as mentioned by @shazqadeer offline we may want to get this diff in without necessarily having this line.
I understand that has the problem of how to test this code and how functional tests can be made to exercise a more exhaustive usage of the verifier. 
One possibility would be to plug this in the usage of the verifier for the compiler but not on chain.
Not sure, those are just thoughts...

Reply:
I think what you suggested is perfectly suitable for the unused entry checker, but not this one.

Instantiation loop is sth we need to rule out, either by this static pass, or some other mechanism at the runtime. The later does not exist yet, so I'd say for the correctness of generics we should run this on chain (at least for now).

Reply:
to be honest we could easily go down the path of leaving the check out for now, but I am fine putting it in and if we decide otherwise we change it, so it's all good.
Overall there is a lot of value in this check and if not enabling it right away makes it easier to get it in let's do that. I am saying that just as a future reference for similar work

Reply:
It's the opposite: not enabling it requires disabling the tests.